util/util.go

```diff
@@ -1,15 +1,82 @@
 package util
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"os"
 	"strings"
+	"time"
 	"unicode"
 	"unicode/utf8"
 
 	"github.com/algorand/go-codec/codec"
+
+	log "github.com/sirupsen/logrus"
 )
 
+// ErrTimeout is returned when CallWithTimeout has a normal timeout.
+var ErrTimeout = errors.New("timeout during call")
+
+// IsTimeoutError compares the given error against the timeout errors.
+func IsTimeoutError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return errors.Is(err, ErrTimeout)
+}
+
+// ErrMisbehavingHandler is written to the log when a handler does not return.
+var ErrMisbehavingHandler = "Misbehaving handler did not exist after 1 second."
+
+// misbehavingHandlerDetector warn if ch does not exit after 1 second.
+func misbehavingHandlerDetector(log *log.Logger, ch chan struct{}) {
+	if log == nil {
+		return
+	}
+
+	select {
+	case <-ch:
+		// Good. This means the handler returns shortly after the context finished.
+		return
+	case <-time.After(1 * time.Second):
+		log.Warnf(ErrMisbehavingHandler)
```

Comment:
I think there is no need for `ErrMisbehavingHandler`. We can just put the string here.

Reply:
I'll make it private. I added a unit test so wanted a variable to test against.